virt-v2v/cold/entrypoint.go

```diff
@@ -195,33 +195,38 @@ func LinkDisks(diskKind string, num int) (err error) {
 	return
 }
 
-func executeVirtV2v(args []string) (err error) {
-	virtV2vCmd := exec.Command(args[0], args[1:]...)
-	r, w := io.Pipe()
-	virtV2vCmd.Stdout = w
-	virtV2vCmd.Stderr = w
-
-	fmt.Println("exec ", virtV2vCmd)
-	if err = virtV2vCmd.Start(); err != nil {
-		fmt.Printf("Error executing command: %v\n", err)
-		return
+func executeVirtV2v(args []string) error {
+	v2vCmd := exec.Command(args[0], args[1:]...)
+	monitorCmd := exec.Command("/usr/local/bin/virt-v2v-monitor")
+	monitorCmd.Stdout = os.Stdout
+	monitorCmd.Stderr = os.Stderr
+
+	var writer *io.PipeWriter
+	monitorCmd.Stdin, writer = io.Pipe()
+	v2vCmd.Stdout = writer
+	v2vCmd.Stderr = writer
+	defer writer.Close()
+
+	if err := monitorCmd.Start(); err != nil {
+		fmt.Printf("Error executing monitor command: %v\n", err)
+		return err
 	}
 
-	virtV2vMonitorCmd := exec.Command("/usr/local/bin/virt-v2v-monitor")
-	virtV2vMonitorCmd.Stdin = r
-	virtV2vMonitorCmd.Stdout = os.Stdout
-	virtV2vMonitorCmd.Stderr = os.Stderr
-
-	if err = virtV2vMonitorCmd.Start(); err != nil {
-		fmt.Printf("Error executing monitor command: %v\n", err)
-		return
+	fmt.Println("exec:", v2vCmd)
+	if err := v2vCmd.Run(); err != nil {
+		fmt.Printf("Error executing v2v command: %v\n", err)
+		return err
 	}
 
-	if err = virtV2vCmd.Wait(); err != nil {
-		fmt.Printf("Error waiting for virt-v2v to finish: %v\n", err)
-		return
+	// virt-v2v is done, we can close the pipe to virt-v2v-monitor
+	writer.Close()
+
+	if err := monitorCmd.Wait(); err != nil {
```

Comment:
Do we really care if the monitor finished? once virt-v2v done it should be enough of indication the conversion is done successfully, and we also have the mechanism in the controller that know its done even if some of the metrics were missing or wasn't properly sent. I just not sure if it worth to fail or block the migration In case the monitor command facing some issues.

Reply:
we can call Wait on a best-effort basis but we need to call it in order to release resources

Reply:
> we can call Wait on a best-effort basis but we need to call it in order to release resources

I don't remember the details, but we had a case in which the monitoring command didn't finish successfully and it really was an indication of an issue - and as this code exists in the release-v2.6.2 branch and works well, I don't see a reason to ignore the error (on the contrary, if we would start getting an error here, we should rather look into it to see the root cause)